### PR TITLE
Bonus for spell: cast without skip the turn

### LINKED ISSUE #4088: Need a bonus for spell: cast without skip the turn
**Describe your proposal**
Similar as the setting: "canCastOnSelf" : true,

It should be an extra setting in spell format,
like "canCastWithoutSkip", if in true, the creature will not skip the turn.
default for false. 

For mod use, can make the label for modding.

### PR DESCRIPTION
**Pull request by Laserlicht**
Fixes #4088

### CODE REVIEW

File: lib/battle/CUnitState.h [hunk 1]
```diff
@@ -141,6 +141,7 @@ class DLL_LINKAGE CUnitState : public Unit
 	bool drainedMana;
 	bool fear;
 	bool hadMorale;
+	SpellID usedSpell;
```
Comment (IvanSavenko) [inline lib/battle/CUnitState.h:144]: Perhaps simple `castSpellThisTurn` will do? From what I see it is only used as a boolean.

Comment (Laserlicht) [reply lib/battle/CUnitState.h:144]: It's needed as SpellID in BattleFlowProcessor because functionality based on spell config.

Comment (Laserlicht) [reply lib/battle/CUnitState.h:144]: It's not. This was my first approach and not working. I've forgotten, why, but it's not the same (it's NONE at time of check).

Comment (IvanSavenko) [reply lib/battle/CUnitState.h:144]: Don't see how this could be possible to be honest...

Comment (Laserlicht) [reply lib/battle/CUnitState.h]: Yes. You're right. My mistake.

**Pull request merged**